src/master-playlist-controller.js

```diff
@@ -544,7 +545,23 @@ export class MasterPlaylistController extends videojs.EventTarget {
     this.mainSegmentLoader_.on('bandwidthupdate', () => {
       // figure out what stream the next segment should be downloaded from
       // with the updated bandwidth information
-      this.masterPlaylistLoader_.media(this.selectPlaylist());
+      const nextPlaylist = this.selectPlaylist();
+      const currentPlaylist = this.masterPlaylistLoader_.media();
+      const buffered = this.tech_.buffered();
+      const forwardBuffer = buffered.length ?
+        buffered.end(buffered.length - 1) - this.tech_.currentTime() : 0;
+
+      // we want to switch down to lower resolutions quickly to continue playback, but
```

Comment:
We can just add "For VOD" here, and below we can say "For live, we want to avoid using a low water line, as we only have 3 segment durations worth of content to buffer."